generators/app/templates/auth0.ejs

```diff
@@ -0,0 +1,29 @@
+const jwt = require('express-jwt');
+const jwksRsa = require('jwks-rsa');
+<% if(scopeValidation) {%>const jwtAuthz = require('express-jwt-authz');<%}%>
+const config = require('../../config');
+
+const { auth0endpoint, auth0audience } = config.common.auth0;
+
+module.exports =
+  config.isTesting || config.isDevelopment
+    ? (req, res, next) => next()
+    : jwt({
+        secret: jwksRsa.expressJwtSecret({
+          cache: true,
+          rateLimit: true,
+          jwksRequestsPerMinute: 5,
+          jwksUri: `${auth0endpoint}/.well-known/jwks.json`
+        }),
+
+        audience: `${auth0audience}`,
+        issuer: `${auth0endpoint}/`,
+        algorithms: ['RS256']
+      });
+<% if(scopeValidation) {%><% if(comments) {%>
+// In general the scopes are defined with this format:
+// permission:resource
+// where "permission" is the permission level that the user that called the endpoint needs to access this resource using this endpoint. It could be for example: read, delete, update, create, register, view, etc...
+// and "resource" is the resourse for which you request the permission.<%}%>
+exports.checkScopes =
+  config.isTesting || config.isDevelopment ? (req, res, next) => next() : jwtAuthz(['read:messages']);<%}%>
```

Comment:
Environment should not change behaviour of this (that could be done with a mock), also would be nice pass the permissions to this function (like `exports.checkScopes = permissions => ....`)